taskfile/node_remote.go

```diff
@@ -0,0 +1,345 @@
+	*BaseNode
+
+	cachedSource     *source
+	client           *getter.Client
+	logger           *logger.Logger
+	proto            string
+	taskfileOverride string
+	timeout          time.Duration
+	url              *url.URL
+}
+
+func NewRemoteNode(
+	l *logger.Logger,
+	entrypoint string,
+	dir string,
+	insecure bool,
+	timeout time.Duration,
+	opts ...NodeOption,
+) (*RemoteNode, bool, error) {
+	client := newGetterClient(dir)
+	proto, u, err := extractProtocolFromURL(client, entrypoint)
+	if err != nil {
+		return nil, false, fmt.Errorf("error parsing remote protocol %s: %w", entrypoint, err)
+	}
+
+	if proto == "file" {
+		// We don't support go-getter's file implementation because it doesn't give direct access to the file,
+		// returning a symlink instead.  We prefer to rely on our own implementation.
+		return nil, false, nil
+	}
+
+	u, err = resolveHTTPEntrypoint(l, insecure, timeout, u)
+	if err != nil {
+		return nil, false, err
+	}
+
+	var tf string
+	tf, u = resolveTaskfileOverride(u)
+
+	return &RemoteNode{
+		BaseNode:         NewBaseNode(dir, opts...),
+		client:           client,
+		logger:           l,
+		proto:            proto,
+		taskfileOverride: tf,
+		timeout:          timeout,
+		url:              u,
+	}, true, nil
+}
+
+func (r *RemoteNode) Location() string {
+	return r.proto + "::" + r.url.String()
+}
+
+func (r *RemoteNode) Read() (*source, error) {
+	return r.loadSource()
+}
+
+func (r *RemoteNode) ResolveEntrypoint(entrypoint string) (string, error) {
+	childProto, _, err := extractProtocolFromURL(r.client, entrypoint)
+	if err != nil {
+		return "", fmt.Errorf("could not resolve protocol for include %s: %w", entrypoint, err)
+	}
+
+	if childProto == "file" && !filepath.IsAbs(entrypoint) {
+		// Relative file paths are resolved as relative to our own source location
+		src, err := r.loadSource()
+		if err != nil {
+			return "", err
+		}
+
+		relativePath := filepath.Join(src.FileDirectory, entrypoint)
+		if exists, err := fileExists(relativePath); err != nil {
+			return "", err
+		} else if exists {
+			return relativePath, nil
+		}
+
+		if r.proto == "http" || r.proto == "https" {
+			// In HTTP, if relative includes are not available locally (eg. from a ZIP file),
+			// we try to download them relative to the base URL.
+			rel, err := url.Parse(entrypoint)
+			if err != nil {
+				return "", fmt.Errorf("error parsing entrypoint %s as url: %w", entrypoint, err)
+			}
+
+			return r.url.ResolveReference(rel).String(), nil
+		}
+	}
+
+	return entrypoint, nil
+}
+
+func (r *RemoteNode) ResolveDir(dir string) (string, error) {
+	path, err := execext.Expand(dir)
+	if err != nil {
+		return "", err
+	}
+
+	if filepathext.IsAbs(path) {
+		return path, nil
+	}
+
+	// NOTE: Uses the directory of the entrypoint (Taskfile), not the current working directory
+	// This means that files are included relative to one another
+	entrypointDir := filepath.Dir(r.Dir())
+	return filepathext.SmartJoin(entrypointDir, path), nil
```

Comment:
I'm not sure if this is correct.  I am asking about it in https://github.com/go-task/task/pull/1757#issuecomment-2341545646.